src/core/analysis-context.ts

```diff
@@ -247,16 +247,26 @@ export class AnalysisContext {
   }
 
   /**
-   * Return a copy, but with the given cache.
+   * Returns a copy of the context but with optional replacements of cache or
+   * constructor options.
+   *
+   * Note: this feature is experimental.
    */
-  private _fork(cache: AnalysisCache): AnalysisContext {
-    const copy = new AnalysisContext({
+  _fork(cache?: AnalysisCache, options?: ForkOptions): AnalysisContext {
+    const contextOptions: Options = {
       lazyEdges: this._lazyEdges,
       parsers: this._parsers,
       scanners: this._scanners,
       urlLoader: this._loader,
-      urlResolver: this._resolver
-    });
+      urlResolver: this._resolver,
+    };
+    if (options) {
+      Object.assign(contextOptions, options);
+    }
+    const copy = new AnalysisContext(contextOptions);
+    if (!cache) {
+      cache = this._cache.invalidate([]);
+    }
     copy._cache = cache;
```

Comment:
Perhaps the correct thing to do now is just to limit the override-able settings to the `urlLoader` since at least in this case, the analyzer's behavior is predictable.  Dealing with the other cases seems like they need some special handling and explanation that is beyond what should be tested for here.  I've updated the types/methods/tests accordingly.